diff --git a/core/misc/ajax.es6.js b/core/misc/ajax.es6.js
index 6eeddf3df7..635952409d 100644
--- a/core/misc/ajax.es6.js
+++ b/core/misc/ajax.es6.js
@@ -1548,7 +1548,7 @@
      *   The XMLHttpRequest status.
      */
     add_css(ajax, response, status) {
-      $('head').prepend(response.data);
+      $('head').append(response.data);
     },
 
     /**
diff --git a/core/misc/ajax.js b/core/misc/ajax.js
index 9135ae485c..e904a458a2 100644
--- a/core/misc/ajax.js
+++ b/core/misc/ajax.js
@@ -636,7 +636,7 @@ function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len
       $("input[name=\"form_build_id\"][value=\"".concat(response.old, "\"]")).val(response.new);
     },
     add_css: function add_css(ajax, response, status) {
-      $('head').prepend(response.data);
+      $('head').append(response.data);
     },
     message: function message(ajax, response) {
       var messages = new Drupal.Message(document.querySelector(response.messageWrapperQuerySelector));
