diff --git a/core/modules/media/js/plugins/drupalmedia/plugin.es6.js b/core/modules/media/js/plugins/drupalmedia/plugin.es6.js
index ef7ea3f9b5..c177544810 100644
--- a/core/modules/media/js/plugins/drupalmedia/plugin.es6.js
+++ b/core/modules/media/js/plugins/drupalmedia/plugin.es6.js
@@ -222,6 +222,7 @@
               widget._setUpDynamicEditables();
               widget._setUpEditButton();
               editor.fire('unlockSnapshot');
+              editor.fire('saveSnapshot');
             });
           }
 
diff --git a/core/modules/media/js/plugins/drupalmedia/plugin.js b/core/modules/media/js/plugins/drupalmedia/plugin.js
index c26512e732..46c43fe773 100644
--- a/core/modules/media/js/plugins/drupalmedia/plugin.js
+++ b/core/modules/media/js/plugins/drupalmedia/plugin.js
@@ -172,6 +172,7 @@
               widget._setUpEditButton();
 
               editor.fire('unlockSnapshot');
+              editor.fire('saveSnapshot');
             });
           }
 
@@ -329,4 +330,4 @@
       linkCommandIntegrator(editor);
     }
   });
-})(jQuery, Drupal, CKEDITOR);
\ No newline at end of file
+})(jQuery, Drupal, CKEDITOR);
