diff --git a/modules/config_distro_ignore/src/Controller/ConfigDistroIgnoreController.php b/modules/config_distro_ignore/src/Controller/ConfigDistroIgnoreController.php
index a3131426615e226c9e5000b4e388f7cbfcb9e4e2..410d968efd7e1ab1b740fb383f452d90591c1d39 100644
--- a/modules/config_distro_ignore/src/Controller/ConfigDistroIgnoreController.php
+++ b/modules/config_distro_ignore/src/Controller/ConfigDistroIgnoreController.php
@@ -10,7 +10,7 @@ use Symfony\Component\DependencyInjection\ContainerInterface;
 /**
  * Returns responses for config module routes.
  */
-class ConfigDistroIgnoreController implements ContainerInjectionInterface {
+class ConfigDistroIgnoreController extends ConfigController implements ContainerInjectionInterface {
 
   /**
    * {@inheritdoc}
