diff --git a/src/Controller/LogoutController.php b/src/Controller/LogoutController.php
index f8ba2f1..c14c62a 100644
--- a/src/Controller/LogoutController.php
+++ b/src/Controller/LogoutController.php
@@ -71,7 +71,7 @@ class LogoutController implements ContainerInjectionInterface {
    * {@inheritdoc}
    */
   public static function create(ContainerInterface $container) {
-    return new static($container->get('cas.helper'), $container->get('request_stack'), $container->get('config.factory'), $container->get('url_generator.non_bubbling'));
+    return new static($container->get('cas.helper'), $container->get('request_stack'), $container->get('config.factory'), $container->get('url_generator'));
   }
 
   /**
